docs/IE_PLUGIN_DG/Plugin.md

```diff
@@ -30,152 +30,141 @@ Based on that, declaration of a plugin class can look as follows:
-> configuration set via `Plugin::SetConfig`, where some values are overwritten with `config` passed to `Plugin::LoadExeNetworkImpl`. 
-> Therefore, the config of  `Plugin::LoadExeNetworkImpl` has a higher priority.
+Before a creation of an `CompiledModel` instance via a constructor, a plugin may check if a provided 
+ov::Model object is supported by a device if it is needed.
 
-### `TransformNetwork()`
+Actual graph compilation is done in the `CompiledModel` constructor. Refer to the [CompiledModel Implementation Guide](@ref openvino_docs_ie_plugin_dg_executable_network) for details.
 
-The function accepts a const shared pointer to `ov::Model` object and performs the following steps:
+> **NOTE**: Actual configuration map used in `CompiledModel` is constructed as a base plugin 
+> configuration set via `Plugin::set_property`, where some values are overwritten with `config` passed to `Plugin::compile_model`. 
+> Therefore, the config of  `Plugin::compile_model` has a higher priority.
 
-1. Deep copies a const object to a local object, which can later be modified.
-2. Applies common and plugin-specific transformations on a copied graph to make the graph more friendly to hardware operations. For details how to write custom plugin-specific transformation, please, refer to [Writing OpenVINO™ transformations](@ref openvino_docs_transformations) guide. See detailed topics about network representation:
+### `transform_model()`
+
+The function accepts a const shared pointer to `ov::Model` object and applies common and plugin-specific transformations on a copied graph to make the graph more friendly to hardware operations. For details how to write custom plugin-specific transformation, please, refer to [Writing OpenVINO™ transformations](@ref openvino_docs_transformations) guide. See detailed topics about network representation:
     * [Intermediate Representation and Operation Sets](@ref openvino_docs_MO_DG_IR_and_opsets)
-    * [Quantized networks](@ref openvino_docs_ie_plugin_dg_quantized_networks).
+    * [Quantized models](@ref openvino_docs_ie_plugin_dg_quantized_networks).
 
-@snippet template/src/plugin.cpp plugin:transform_network
+@snippet template/src/plugin.cpp plugin:transform_model
 
-> **NOTE**: After all these transformations, a `ov::Model` object contains operations which can be perfectly mapped to backend kernels. E.g. if backend has kernel computing `A + B` operations at once, the `TransformNetwork` function should contain a pass which fuses operations `A` and `B` into a single custom operation `A + B` which fits backend kernels set.
+> **NOTE**: After all these transformations, a `ov::Model` object contains operations which can be perfectly mapped to backend kernels. E.g. if backend has kernel computing `A + B` operations at once, the `transform_model` function should contain a pass which fuses operations `A` and `B` into a single custom operation `A + B` which fits backend kernels set.
 
-### `QueryNetwork()`
+### `query_model()`
```

Comment:
for some reason I don't see this header in the docs, as well as transform_model, compile_model and others